agent/xds/clusters.go

```diff
@@ -94,6 +93,11 @@ func (s *ResourceGenerator) clustersFromSnapshotConnectProxy(cfgSnap *proxycfg.C
 			return nil, fmt.Errorf("no endpoint map for upstream %q", uid)
 		}
 
+		if _, ok := cfgSnap.ConnectProxy.WatchedUpstreamTrustBundles[uid.Peer]; uid.Peer != "" && !ok {
+			// The trust bundle for this upstream is not available yet, skip for now.
+			continue
+		}
```

Comment:
This conditional skip needs to go into the EDS loop as well to keep clusters and endpoints in sync.